libtenzir/builtins/operators/hash.cpp

```diff
@@ -193,8 +195,37 @@ class plugin final : public virtual operator_plugin<hash_operator> {
   }
 };
 
+class plugin2 : public virtual function_plugin {
+  auto name() const -> std::string override {
+    return "xxhash3";
+  }
+
+  auto make_function(invocation inv,
+                     session ctx) const -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    auto salt = std::optional<std::string>{};
+    TRY(argument_parser2::function("xxhash3")
+          .add(expr, "<field>")
+          .add(salt, "[salt]")
```

Comment:
The `[...]` is not helpful. Look at this output:
~~~
error: named argument `foo` does not exist
 --> <input>:1:24
  |
1 | from {x: xxhash3("42", foo=bar)}
  |                        ^^^ 
  |
  = usage: xxhash3(<field>, [salt]?)
  = docs: https://docs.tenzir.com/functions/xxhash3
~~~
Optionality is already denoted by the argument parser with `?`.

On a related note: I think `salt` should better be a named argument. Doesn't make sense for me as a positional.

Reply:
Is this also really a "salt"? I would have expected "seed" here. A *salt* is something used in a crypto context, and XXH3 is actually a non-cryptographic hash function.

Reply:
It is a seed, yes. But the original operator called it `salt` so I took the same name.

Reply:
Okay, let's go with `seed` here.